### ISSUE TITLE
It is possible to add an entity from outside the Prefab into it while the Prefab is being edited which may crash the Editor 

### ISSUE THREAD

**Issue by LB-TomaszSerwanski** [o3de/o3de#4858]
**Describe the bug**
It is possible to drag and drop entities from outside the Prefab while editing it after creating a new entity. This allows the User to insert the non-Prefab entities into the Prefab which crashes the Editor. 

Please note that the crash only occurs when the entity is inserted as a top-level child (a child of a prefab container). Otherwise only the following assert appears:

```
(System) - Trace::Assert
 D:/Git/o3de/Code/Framework/AzToolsFramework/AzToolsFramework/UI/Outliner/EntityOutlinerListModel.cpp(147): (9184) 'class QModelIndex __cdecl AzToolsFramework::EntityOutlinerListModel::index(int,int,const class QModelIndex &) const'
(System) - No child found for parent
```

Please refer to the attached video and log files for more details.

**Steps to reproduce**
1. Open any level.
2. Create an entity and make a Prefab out of it.
3. Create another entity outside the Prefab.
4. Double click the Prefab (or RMB -> _Edit Prefab_).
5. RMB the Entity Outliner and select _Create Entity_.
6. Drag and drop the entity from outside the Prefab in between the two entities inside the Prefab (make sure to drop it on the same level as the two entities and not as a child of one of them).

**Expected behavior**
It is not possible to insert entities from outside the Prefab into the Prefab while editing it. The Editor remains stable.

**Actual behavior**
It is possible to insert entities from outside the Prefab into the Prefab while editing it. The Editor crashes.

**Call Stack Trace**
```
 7) 00007FFCD77FC347 (KERNELBASE) : UnhandledExceptionFilter
 6) 00007FFCDA0151B0 (ntdll) : memset
 5) 00007FFCD9FFC766 (ntdll) : _C_specific_handler
 4) 00007FFCDA0120CF (ntdll) : _chkstk
 3) 00007FFCD9FC1454 (ntdll) : RtlRaiseException
 2) 00007FFCDA010BFE (ntdll) : KiUserExceptionDispatcher
 1) D:\Git\o3de\Code\Framework\AzCore\AzCore\IO\Path\Path.inl (837) : AZ::IO::BasicPath<AZStd::basic_string<char,AZStd::char_traits<char>,AZStd::allocator> >::operator/=
```

**Assets required**
N/A

**Screenshots/Video**

https://user-images.githubusercontent.com/86953108/138251826-2632e739-df55-41c4-9f4f-d100dcef6272.mp4

**Found in Branch**
Development (74d7405).

**Desktop/Device:**
 - Device: PC
 - OS: Windows
 - Version 10
 - CPU: AMD Ryzen 5 3600
 - GPU: Nvidia RTX 2060 SUPER
 - Memory: 16GB

**Attachments**
[assert.txt](https://github.com/o3de/o3de/files/7387837/assert.txt)
[Editor.log](https://github.com/o3de/o3de/files/7387838/Editor.log)
[error.log](https://github.com/o3de/o3de/files/7387839/error.log)
[error.zip](https://github.com/o3de/o3de/files/7387840/error.zip)

**Comment by AMZN-daimini** [comment]
Hey Tomasz, the operation you describe is meant to be a valid workflow. Of course the editor should not crash though :D

The other assert you described ("No child found for parent") is actually a separate issue and is being addressed:
https://github.com/o3de/o3de/issues/4798

Will look into this one, thanks :D

**Comment by LB-TomaszSerwanski** [comment]
Fix verified on Stabilization_2110 (a0ab592).
